contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -499,6 +499,62 @@ library VaultLifecycleSTETH {
         }
     }
 
+    /**
+     * @notice Gets stETH for direct stETH withdrawals, converts wstETH/ETH to stETH if not enough stETH
+     * @param steth is the address of steth
+     * @param wstEth is the address of wsteth
+     * @param amount is the amount to withdraw
+     * @return amount of stETH to transfer to the user, this is to account for rounding errors when unwrapping wstETH
+     */
+    function withdrawStEth(
+        address steth,
+        address wstEth,
+        uint256 amount
+    ) external returns (uint256) {
+        // 3 different scenarios for withdrawing stETH directly
+        // Scenario 1. We hold enough stETH to satisfy withdrawal. Send it out directly
+        // Scenario 2. We hold enough stETH + wstETH to satisy withdrawal. Unwrap wstETH then send it
+        // Scenario 3. We hold enough stETH + wstETH + ETH satisfy withdrawal. Unwrap wstETH, wrap ETH then send it
+        uint256 stethBalance = IERC20(steth).balanceOf(address(this));
+        if (stethBalance >= amount) {
+            // Can send out the stETH directly
+            return amount; // We return here if we have enough stETH to satisfy the withdrawal
+        } else {
+            // If amount > stethBalance, send out the entire stethBalance and check wstETH and ETH
+            amount = amount.sub(stethBalance);
+        }
+        uint256 wstethBalance = IWSTETH(wstEth).balanceOf(address(this));
+        stethBalance = IWSTETH(wstEth).getStETHByWstETH(wstethBalance);
+        if (stethBalance >= amount) {
+            wstethBalance = IWSTETH(wstEth).getWstETHByStETH(amount);
+            // Avoids reverting if unwrap amount is 0
+            if (wstethBalance > 0) {
+                // Unwraps wstETH and sends out the received stETH directly
+                IWSTETH(wstEth).unwrap(wstethBalance);
+                // Accounts for rounding errors when unwrapping wstETH, this is safe because this function would've
+                // returned already if the stETH balance was greater than our withdrawal amount
+                return IERC20(steth).balanceOf(address(this)); // We return here if we have enough stETH + wstETH
+            }
+        } else if (stethBalance > 0) {
+            stethBalance = IERC20(steth).balanceOf(address(this));
+            IWSTETH(wstEth).unwrap(wstethBalance);
+            // Accounts for rounding errors when unwrapping wstETH
+            amount = amount.sub(
+                IERC20(steth).balanceOf(address(this)).sub(stethBalance)
+            );
+        }
+        // Wrap ETH to stETH if we don't have enough stETH + wstETH
+        uint256 ethBalance = address(this).balance;
+        if (amount > 0 && ethBalance >= amount) {
+            ISTETH(steth).submit{value: amount}(address(this));
+        } else if (ethBalance > 0) {
+            ISTETH(steth).submit{value: ethBalance}(address(this));
+        }
```

Comment:
Should we add an `else` statement here and revert?

Reply:
I think having this require check is enough: https://github.com/ribbon-finance/ribbon-v2/pull/214#discussion_r805522611